### DISCUSSION TITLE
CLI obtains complete quoted commands

### DISCUSSION THREAD

**Discussion by heiyelideheideng** [daniele77/cli#246] [General]
hi daniele
Thank you very much for providing the CLI project. I have a requirement to obtain the entire string of the terminal input command, including quotation marks and spaces, without adding additional quotation marks externally. For example, foo hello '123' is the command, followed by two string parameters. '123' contains single quotation marks stored in one string. So I would like to ask if there is any way to obtain the entire string containing spaces and quotation marks in the input command of rootMenu without adding extra quotation marks externally.
Thank you again for your efforts and support!

**Comment by daniele77** [comment]
Hi @heiyelideheideng,

Thank you for your feedback.

Currently, there is no way to retrieve the input string from a command handler. However, this could be a valuable feature to add. In the next release, we plan to pass a UserSession object to every command handler, which could be a good candidate for including the raw request string.

Please add an issue if you are interested in this feature.
